File: Data Structures & Algorithms/duplicate-integer/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let mapp = {};
        for (let i=0 ; i<nums.length ; i++){
            mapp[nums[i]] = (mapp[nums[i]] || 0 ) + 1  
            if (mapp[nums[i]] > 1){
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-3.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int c = 0, m=0;
        for(int& n: nums){
            if (c==0){
                c = n;
                m=max(c,m);
                continue;
            }
            if (c < 0){
                c = n;
                m=max(m,c);
                continue;
            }
            c+= n;
            m= max(m,c);
        }

        return c < 0 ? -1 : m;
    }
};
